CCD-support/Multitenancy/environment.md

```diff
@@ -0,0 +1,552 @@
+else
+  echo "cannot find _clouds_yaml"
+  exit 1
+fi
+
+if [ "$1" == "baremetal" ] ; then
+  shift 1
+fi
+
+# shellcheck disable=SC2086
+sudo podman run --net=host --tls-verify=false \
+  -v "${MOUNTDIR}:/etc/openstack" --rm \
+  -e OS_CLOUD="${OS_CLOUD:-metal3}" "172.22.0.1:5000/localimages/ironic-client" "$@"
+EOF
+
+mkdir _clouds_yaml
+
+cat <<'EOF' > _clouds_yaml/clouds.yaml
+clouds:
+  metal3:
+    auth_type: none
+    baremetal_endpoint_override: http://172.22.0.2:6385
+    baremetal_introspection_endpoint_override: http://172.22.0.2:5050
+EOF
+sudo ln -sf "/home/metal3ci/ironicclient.sh" "/usr/local/bin/baremetal"
+```
+## Run BMO
+1. install cert manager
+```
+wget https://github.com/cert-manager/cert-manager/releases/download/v1.5.3/cert-manager.yaml
+kubectl apply -f cert-manager.yaml
+```
+From the manifest [bmo-wns.yaml](bmo-wns.yaml)
+
+2. Apply bmo in two different ns:
+```
+kubectl apply -f bmo-1.yaml -n baremetal-operator-system-test1
+kubectl apply -f bmo-2.yaml -n baremetal-operator-system-test2
+kubectl create ns test1
+kubectl create ns test2
+kubectl apply -f role.yaml -n test1
+kubectl apply -f role-binding.yaml -n test1
+kubectl apply -f role-2.yaml -n test2
+kubectl apply -f role-binding-2.yaml -n test
+```
+3. Creating bmhs
+```
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: node-0-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: node-0
+spec:
+  online: true
+  bootMACAddress: 00:5c:52:31:3a:9c
+  bootMode: legacy
+  bmc:
+    address: ipmi://192.168.111.1:6230
+    credentialsName: node-0-bmc-secret
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: node-1-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+metadata:
+  name: node-1
+spec:
+  online: true
+  bootMACAddress: 00:5c:52:31:3a:ad
+  bootMode: legacy
+  bmc:
+    address: ipmi://192.168.111.1:6231
+    credentialsName: node-1-bmc-secret
+
+```
+```
+kubectl get bmh -A
+```
+```
+NAMESPACE   NAME     STATE          CONSUMER                   ONLINE   ERROR   AGE
+test1       node-1   available                                  true            3m
+test2       node-2   available                                  true            2m
+
+```
+## Init the management cluster 
+1. Get clusterctl
+```
+curl -L https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.1.5/clusterctl-linux-amd64 -o clusterctl
+chmod +x ./clusterctl
+sudo mv ./clusterctl /usr/local/bin/clusterctl
+```
+2. Run capi/capm3
+```
+clusterctl init --core cluster-api:v1.1.5 --bootstrap kubeadm:v1.1.5 --control-plane kubeadm:v1.1.5 --infrastructure=metal3:v1.1.2 -v5
+```
+
+## Apply cluster templates for each namespace:
+make sure the netwrok are correct for each and use different cluster endpoint ip for each cluster since they still share the same baremetal network
```

Comment:
```suggestion
Make sure the networks are correct for each and use different cluster endpoint IP for each cluster since they still share the same baremetal network
```